cluster/log-dump/log-dump.sh

```diff
@@ -429,6 +429,30 @@ function dump_nodes_with_logexporter() {
   fi
 }
 
+function detect_node_failures() {
+  if ! [[ "${gcloud_supported_providers}" =~ "${KUBERNETES_PROVIDER}" ]]; then
+    return
+  fi
+
+  detect-node-names
+  for group in "${INSTANCE_GROUPS[@]}"; do
+    local creation_timestamp=$(gcloud compute instance-groups managed describe \
+                              "${group}" \
+                              --project "${PROJECT}" \
+                              --zone "${ZONE}" \
+                              --format='value(creationTimestamp)')
+    echo "Failures for ${group}"
+    gcloud logging read --order=asc \
+          --format='value(timestamp,jsonPayload.resource.name,jsonPayload.event_subtype)' \
+          --project "${PROJECT}" \
+          "resource.type=\"gce_instance\"
+           logName=\"projects/${PROJECT}/logs/compute.googleapis.com%2Factivity_log\"
+           (jsonPayload.event_subtype=\"compute.instances.hostError\" OR jsonPayload.event_subtype=\"compute.instances.automaticRestart\")
+           jsonPayload.resource.name:\"${group}\"
+           timestamp >= \"${creation_timestamp}\""
```

Comment:
This looks great.
My only question here is - what is the estimation on the time it takes to run this command? This pretty much goes over activity log on all nodes, which for 5k-node clusters may be quite a lot.
Did you try running that on larger clusters?

Reply:
It takes ~20 seconds per instance group. So approx. 1 minute for all of them.

Reply:
Cool - thanks.